lib/tokenize.cpp

```diff
@@ -10114,8 +10114,13 @@ void Tokenizer::simplifyKeyword()
             tok->deleteNext();
 
         if (c99) {
-            while (tok->str() == "restrict")
-                tok->deleteThis();
+            if (tok->str() == "restrict") {
+                // check if restrict is used properly
+                if ((tok->strAt(-1) == "*" && tok->next()->isName()) || tok->strAt(-1) == "[" || tok->strAt(1) == "*")
+                    tok->deleteThis();
+                else
+                    syntaxErrorC(tok, "restrict");
```

Comment:
In your first proposal there would not be a syntax error. Somehow I like that better.

It seems a bit fragile to treat each location uniquely though.. it would be more robust to first iterate through the whole TU and determine if `restrict` is a keyword or identifier. What do you think?

I believe we could do that for some other keywords also. instead of treating each location separately we treat them all together. but that is not for this PR of course.

Reply:
If we don't throw a syntax error then it would be nice if this code work:

    void f(int restrict) {
        a = buf[restrict];
    }

Reply:
I'll redo and resubmit.